common/src/main/java/org/ergoplatform/appkit/Eip29Attachment.java

```diff
@@ -0,0 +1,221 @@
+package org.ergoplatform.appkit;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import scala.Tuple2;
+import special.collection.Coll;
+
+/**
+ * Represents an attachment according to EIP-29
+ */
+public class Eip29Attachment {
+    public static final byte[] MAGIC_BYTES = new byte[]{0x50, 0x52, 0x50};
+    private final int attachmentType;
+    private final byte[] attachmentContent;
+
+    private Eip29Attachment(int attachmentType, byte[] attachmentContent) {
+        this.attachmentType = attachmentType;
+        this.attachmentContent = attachmentContent;
+    }
+
+    /**
+     * @return type of this attachment, if known
+     */
+    public Type getType() {
+        return Eip29Attachment.Type.fromTypeRawValue(attachmentType);
+    }
+
+    /**
+     * @return raw value type of this attachment. Can return types unknown to this implementation.
+     */
+    public int getTypeRawValue() {
+        return attachmentType;
+    }
+
+    /**
+     * @return returns full ErgoValue for this attachment, to use for register 9 of out boxes
+     */
+    public ErgoValue<Tuple2<byte[], Tuple2<Integer, byte[]>>> getErgoValue() {
+        ErgoValue<Tuple2> contentPair = ErgoValue.pairOf(ErgoValue.of(getTypeRawValue()), ErgoValue.of(attachmentContent));
```

Comment:
The actual type is ErgoValue<Tuple2<Coll, Tuple2<Integer, Coll>>> due to either a bug or a wrong declaration in ErgoValue class, I had omitted the full declarations to not run into problems with that. I have now changed ErgoValue declarations to the actual Integer type used, but maybe you should take a look at it yourself. Other possible solution is to change the actual content of the ErgoValue, but this might introduce problems with client code using it and additionally, I have not managed to get scala.Int to be used.